src/parser/ast.rs

```diff
@@ -549,13 +549,19 @@ impl<'a> Ast<'a> {
     }
 
     pub fn if_expr(condition: Ast<'a>, then: Ast<'a>, otherwise: Option<Ast<'a>>, location: Location<'a>) -> Ast<'a> {
-        Ast::If(If {
+        
+ 
+        match &otherwise{
+            Some(_ast) => Ast::If(If {
             condition: Box::new(condition),
             then: Box::new(then),
             otherwise: otherwise.map(Box::new),
             location,
             typ: None,
-        })
+        }),
+            None => Ast::if_expr(condition,then, Some(Ast::unit_literal(location)), location)
```

Comment:
If you're trying to do the desugaring, you're going about this a bit differently than you should. Going this route for example you should only need to change this function - you don't need to touch type checking at all.

There's also an issue with what you're desugaring too currently. Currently in the None case you're desugaring `if a then b` into `if a then b else ()` while you should be desugaring to `if a then Some b else None` instead.

Reply:
Gotcha! Again thanks for the guidance and patience 😅 ! 
> Currently in the None case you're desugaring if a then b into if a then b else () while you should be desugaring to if a then Some b else None instead.
Would this mean I need to modify the `struct If` in parser/ast.rs such that `pub then: Option<Box<Ast<'a>>>` to allow `then` to be Optional (given in your response, we'd want to Desugar `Some b`)?

Reply:
No - I think you're having some confusion here between the language the compiler is written in - rust - and the language we want to compile - Ante. From the compiler's perspective the `If` struct should represent an if expression in ante. The `then` field is the Ast of then branch, and the `otherwise` field is optional (currently) because it holds the Ast of the else branch which currently isn't required. Ie. it is valid to write `if a then b` with no else. The `then` field isn't optional because `if a else b` is not valid syntax, the then branch must always be present. It is just now we want to wrap the whole branch in this call to `Some`. So the branch is still there, there's just an extra function call in it now.
What we could actually do is the reverse - since we'd now always be desugaring `if a then b` to `if a then Some b else None`, we now always have an else branch in every if. So the `otherwise` field is not optional anymore.